internal/engine/wazevo/backend/isa/amd64/machine.go

```diff
@@ -398,6 +398,325 @@ func (m *machine) LowerInstr(instr *ssa.Instruction) {
 		m.lowerVbandnot(instr, sseOpcodePandn)
 	case ssa.OpcodeVbitselect:
 		m.lowerVbitselect(instr)
+	case ssa.OpcodeVIadd:
+		x, y, lane := instr.Arg2WithLane()
+		rn := m.getOperand_Mem_Reg(m.c.ValueDefinition(x))
+		rm := m.getOperand_Reg(m.c.ValueDefinition(y))
+		rd := m.c.VRegOf(instr.Return())
+
+		var vecOp sseOpcode
+		switch lane {
+		case ssa.VecLaneI8x16:
+			vecOp = sseOpcodePaddb
+		case ssa.VecLaneI16x8:
+			vecOp = sseOpcodePaddw
+		case ssa.VecLaneI32x4:
+			vecOp = sseOpcodePaddd
+		case ssa.VecLaneI64x2:
+			vecOp = sseOpcodePaddq
+		}
+
+		tmp := m.copyToTmp(rm.r)
+		i := m.allocateInstr()
+		i.asXmmRmR(vecOp, rn, tmp)
+		m.insert(i)
+
+		m.copyTo(tmp, rd)
+	case ssa.OpcodeVSaddSat:
+		x, y, lane := instr.Arg2WithLane()
+		rn := m.getOperand_Mem_Reg(m.c.ValueDefinition(x))
+		rm := m.getOperand_Reg(m.c.ValueDefinition(y))
+		rd := m.c.VRegOf(instr.Return())
+
+		var vecOp sseOpcode
+		switch lane {
+		case ssa.VecLaneI8x16:
+			vecOp = sseOpcodePaddsb
+		case ssa.VecLaneI16x8:
+			vecOp = sseOpcodePaddsw
+		}
+		tmp := m.copyToTmp(rm.r)
+
+		i := m.allocateInstr()
+		i.asXmmRmR(vecOp, rn, tmp)
+		m.insert(i)
+
+		m.copyTo(tmp, rd)
+	case ssa.OpcodeVUaddSat:
+		x, y, lane := instr.Arg2WithLane()
+		rn := m.getOperand_Mem_Reg(m.c.ValueDefinition(x))
+		rm := m.getOperand_Reg(m.c.ValueDefinition(y))
+		rd := m.c.VRegOf(instr.Return())
+
+		var vecOp sseOpcode
+		switch lane {
+		case ssa.VecLaneI8x16:
+			vecOp = sseOpcodePaddusb
+		case ssa.VecLaneI16x8:
+			vecOp = sseOpcodePaddusw
+		}
+		tmp := m.c.AllocateVReg(ssa.TypeV128)
+		copy := m.allocateInstr()
+		copy.asXmmUnaryRmR(sseOpcodeMovdqu, rn, tmp)
+		m.insert(copy)
```

Comment:
copyToTmp?

Reply:
Rn is mem reg. Copytmp wouldn't handle that right?

Reply:
why not use `getOperand_Mem_Reg` for rm, not rn in the first place? it doesn't make sense

Reply:
(that applies to the other instances as well). I can create a copytotmp that handles operands though (copyOperandToTmp?)

Reply:
I will end up doing the same thing https://github.com/tetratelabs/wazero/commit/2a49278109ba024f88a06dbe850e6d92be152f19 ...

Reply:
ok I think I should have caught all of them now...